### DISCUSSION TITLE
Recommended Authentication method for applications that run as scheduled tasks

### DISCUSSION THREAD

**Discussion by cbarnes01** [pnp/pnpcore#510] [Q&A]
What is the recommended authentication method for applications that run as a scheduled task.   I got the console app sample working with the credential manager, but it looks like the onmicrosoft.com account I created will be forced to use MFA after 14 days.

**Comment by jansenbe** [comment]
@cbarnes01 : recommend option is to use application permissions. For working with SharePoint and Graph this implies setting up an application with a certificate in Azure AD and then using that as described here: https://pnp.github.io/pnpcore/using-the-sdk/configuring%20authentication.html#application-permissions-acting-as-an-app-account-with-app-only-permissions
